shared/src/metabase/query_processor/error_type.cljc

```diff
@@ -3,37 +3,39 @@
   is one of the types here. If you see an Exception in QP code that doesn't return an `:type`, add it!
 
     (throw (ex-info (tru \"Don''t know how to parse {0} {1}\" (class x) x)
-                    {:type qp.error-type/invalid-parameter}))")
+                    {:type qp.error-type/invalid-parameter}))"
+  #?(:cljs (:require-macros [metabase.query-processor.error-type :refer [deferror]])))
 
 (def ^:private hierarchy
-  (make-hierarchy))
+  (atom (make-hierarchy)))
 
```

Comment:
for cljs hierarchies, need an atom. Now clj has an atom too. That seemed far better than reader conditional-ing everything `#?(:cljs @hierarchy :clj hierarchy)` for values and similarly `#?(:cljs hierarchy :clj #'hierarchy)` for the thing to mutate with alter-var-root or atom functions